### DISCUSSION TITLE
Providing different resources for assets in the same definition

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17840] [Q&A] [answered]
If I have 2 assets that are in the same definition, is there a way i can provide a different resource for each asset using the same resource_key?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699478469274819?thread_ts=1699478469.274819&cid=C01U954MEER).*

**Comment by jamiedemaria** [comment, marked as answer]
If the assets have the same resource key, then they can't be provided different resources. You'll need to give the assets different resource keys
